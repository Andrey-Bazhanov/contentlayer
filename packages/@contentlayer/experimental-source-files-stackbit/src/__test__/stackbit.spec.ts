import * as SourceFiles from '@contentlayer/source-files'
import { provideDummyTracing } from '@contentlayer/utils'
import type { HasClock, HasConsole, OT } from '@contentlayer/utils/effect'
import { pipe, provideTestConsole, T } from '@contentlayer/utils/effect'
import * as Stackbit from '@stackbit/sdk'
import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { expect, test } from 'vitest'

import { stackbitConfigToDocumentTypes } from '../index.js'

const testDirPath = fileURLToPath(new URL('.', import.meta.url))
const NOT_USED_STR = 'NOT_USED'

test('dummy', async () => {
  const nextStarterFixtureDirPath = path.join(testDirPath, 'fixtures', 'next-starter')

  const configResult = await Stackbit.loadConfig({ dirPath: nextStarterFixtureDirPath })

  const documentTypes = stackbitConfigToDocumentTypes(configResult.config!)

  const contentlayerSource = await SourceFiles.makeSource({ contentDirPath: NOT_USED_STR, documentTypes })
  const { result: coreSchema } = await runMain(contentlayerSource.provideSchema(NOT_USED_STR))

  expect(coreSchema).toMatchSnapshot()
})

const runMain = async <E, A>(eff: T.Effect<OT.HasTracer & HasClock & HasConsole, E, A>) => {
  const logMessages: string[] = []
  const result = await pipe(eff, provideTestConsole(logMessages), provideDummyTracing, T.runPromise)

  return { logMessages, result }
}
