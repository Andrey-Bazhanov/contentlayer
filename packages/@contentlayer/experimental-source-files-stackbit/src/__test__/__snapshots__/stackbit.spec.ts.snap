// Vitest Snapshot v1

exports[`dummy 1`] = `
{
  "documentTypeDefMap": {
    "Page": {
      "_tag": "DocumentTypeDef",
      "computedFields": [],
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "type",
          "type": "string",
        },
        {
          "default": "This is a new page",
          "description": undefined,
          "isRequired": true,
          "isSystemField": false,
          "name": "title",
          "type": "string",
        },
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "sections",
          "of": [
            {
              "nestedTypeName": "CardGridSection",
              "type": "nested",
            },
            {
              "nestedTypeName": "HeroSection",
              "type": "nested",
            },
          ],
          "type": "list_polymorphic",
          "typeField": "type",
        },
        {
          "default": undefined,
          "description": "Markdown file body",
          "isRequired": true,
          "isSystemField": true,
          "name": "body",
          "type": "markdown",
        },
      ],
      "isSingleton": false,
      "name": "Page",
    },
    "SiteConfig": {
      "_tag": "DocumentTypeDef",
      "computedFields": [],
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": undefined,
          "description": "The type of the object",
          "isRequired": false,
          "isSystemField": false,
          "name": "type",
          "type": "string",
        },
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "title",
          "type": "string",
        },
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "footer",
          "nestedTypeName": "FooterConfig",
          "type": "nested",
        },
        {
          "default": undefined,
          "description": "Markdown file body",
          "isRequired": true,
          "isSystemField": true,
          "name": "body",
          "type": "markdown",
        },
      ],
      "isSingleton": true,
      "name": "SiteConfig",
    },
  },
  "hash": "NOT_USED",
  "nestedTypeDefMap": {
    "Button": {
      "_tag": "NestedTypeDef",
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": "Click Me",
          "description": undefined,
          "isRequired": true,
          "isSystemField": false,
          "name": "label",
          "type": "string",
        },
        {
          "default": "/",
          "description": undefined,
          "isRequired": true,
          "isSystemField": false,
          "name": "url",
          "type": "string",
        },
        {
          "default": "primary",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "theme",
          "options": [
            "primary",
            "secondary",
          ],
          "type": "enum",
        },
      ],
      "name": "Button",
    },
    "Card": {
      "_tag": "NestedTypeDef",
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": "Card Heading",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "heading",
          "type": "string",
        },
        {
          "default": "Card description goes here ...",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "subheading",
          "type": "markdown",
        },
        {
          "default": "/",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "url",
          "type": "string",
        },
      ],
      "name": "Card",
    },
    "CardGridSection": {
      "_tag": "NestedTypeDef",
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": "Card Grid Heading",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "heading",
          "type": "string",
        },
        {
          "default": "Card Grid Subheading",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "subheading",
          "type": "markdown",
        },
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "cards",
          "of": {
            "nestedTypeName": "Card",
            "type": "nested",
          },
          "type": "list",
        },
      ],
      "name": "CardGridSection",
    },
    "FooterConfig": {
      "_tag": "NestedTypeDef",
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "body",
          "type": "markdown",
        },
      ],
      "name": "FooterConfig",
    },
    "HeroSection": {
      "_tag": "NestedTypeDef",
      "description": undefined,
      "extensions": {},
      "fieldDefs": [
        {
          "default": "Hero Heading",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "heading",
          "type": "string",
        },
        {
          "default": "Hero Subheading",
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "subheading",
          "type": "markdown",
        },
        {
          "default": undefined,
          "description": undefined,
          "isRequired": false,
          "isSystemField": false,
          "name": "buttons",
          "of": {
            "nestedTypeName": "Button",
            "type": "nested",
          },
          "type": "list",
        },
      ],
      "name": "HeroSection",
    },
  },
}
`;
